client/context.go

```diff
@@ -229,6 +229,7 @@ func (ctx Context) WithInterfaceRegistry(interfaceRegistry codectypes.InterfaceR
 // client-side config from the config file.
 func (ctx Context) WithViper() Context {
 	v := viper.New()
+	v.AutomaticEnv()
```

Comment:
Also, I'm reading https://github.com/cosmos/cosmos-sdk/issues/8179#issuecomment-747087678. We should actually also allow prefixes, e.g. `GA_CHAIN_ID` or `AKASH_NODE`. What i propose with that WithViper takes a `prefix string` argument, and then we `v.SetEnvPrefix(prefix)`